File: VB/MyTextEditWithCustomAutocomplete/MyTextEditWithCustomAutocomplete.vb
Imports System
Imports DevExpress.Data.Mask
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Mask

Namespace AutoCompleteMaskManager
	<System.ComponentModel.DesignerCategory("")>
	Public Class MyTextEditWithCustomAutocomplete
		Inherits TextEdit

		Protected Overrides Function CreateMaskManager(ByVal mask As MaskProperties) As MaskManager
			Dim maskManager As MaskManager
			If mask.MaskType = MaskType.Custom Then
				maskManager = New MyAutoCompleteMaskManager(New CustomMaskManagerCommon(New MaskManagerPlainTextState(String.Empty, 0, 0)))
			Else
				maskManager = MyBase.CreateMaskManager(mask)
			End If
			Return maskManager
		End Function
	End Class
End Namespace

File: CS/MyTextEditWithCustomAutocomplete/MyTextEditWithCustomAutocomplete.cs
using System;
using DevExpress.Data.Mask;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Mask;

namespace AutoCompleteMaskManager
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyTextEditWithCustomAutocomplete : TextEdit
    {
        protected override MaskManager CreateMaskManager(MaskProperties mask)
        {
            MaskManager maskManager;
            if (mask.MaskType == MaskType.Custom)
                maskManager = new MyAutoCompleteMaskManager(new CustomMaskManagerCommon(new MaskManagerPlainTextState(string.Empty, 0, 0)));
            else
                maskManager = base.CreateMaskManager(mask);
            return maskManager;
        }
    }
}
